pyro/distributions/sine_bivariate_von_mises.py

```diff
@@ -0,0 +1,280 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import warnings
+from copy import copy
+from functools import reduce
+from math import pi
+
+import torch
+from torch.distributions import VonMises
+from torch.distributions.utils import broadcast_all
+
+from pyro.distributions import constraints
+from pyro.distributions.torch_distribution import TorchDistribution
+from pyro.distributions.util import broadcast_shape
+
+
+def _flat_dim(size):
```

Comment:
nit: replace `_flat_dim(shape)` with `shape.numel()` everywhere

Reply:
Thanks; I've was wondering why there wasn't already a method like this.